homeassistant/components/prometheus/__init__.py

```diff
@@ -458,8 +463,12 @@ def _sensor_override_component_metric(self, state, unit):
     def _sensor_fallback_metric(state, unit):
         """Get metric from fallback logic for compatibility."""
         if unit in (None, ""):
-            _LOGGER.debug("Unsupported sensor: %s", state.entity_id)
-            return None
+            try:
+                state_helper.state_as_number(state)
+                return "sensor_state"
```

Comment:
Please move the return below the except block so we only wrap the line that can raise in the try... except block.

Reply:
Thanks, I changed it in the new commit.